### PR TITLE
fix: protogen scripts were not correctly implemented

### LINKED ISSUE #1461: Update protogen scripts to mirror current cosmos-sdk workflow
The protogen scripts and corresponding makefile command targets have to be updated to the current workflow used in the `cosmos-sdk`.

See also [this PR](https://github.com/evmos/evmos/pull/1084) on the Evmos repo.

### PR DESCRIPTION
**Pull request by MalteHerrmann**
Fixes #1461 

To generate the Go implementations of the protobuf interfaces use:
```
make proto-gen
```

To generate the Swagger UI components, use:
```
make proto-swagger-gen
```

Thanks to @itsdevbear, for providing the correct options to use in the `buf.gen.gogo.yaml` files, we were a bit stuck on that before.

### CODE REVIEW

File: scripts/protocgen.sh [hunk 1]
```diff
@@ -2,42 +2,25 @@
...
-}
-
-protoc_gen_doc() {
-  go get -u github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc 2>/dev/null
-} 
-
-protoc_gen_gocosmos
-protoc_gen_doc
+# get cosmos sdk
+go get github.com/cosmos/cosmos-sdk 2>/dev/null
```
Comment (VictorTrustyDev) [inline scripts/protocgen.sh:9]: What is the purpose of this? Without version?

File: proto/buf.yaml [hunk 2]
```diff
@@ -14,6 +20,9 @@ lint:
     - RPC_RESPONSE_STANDARD_NAME
     - RPC_REQUEST_RESPONSE_UNIQUE
     - COMMENT_MESSAGE
+  # TODO: Why do we ignore tendermint here?
+  ignore:
+    - tendermint
```
Comment (fedekunze) [inline proto/buf.yaml:25]: this was to prevent linter errors, but since we're deleting the files we can remove it here too

**Review by fedekunze — COMMENTED** [review]
Looks good. Can you add the instructions to the description and add a Changelog entry?

File: scripts/protocgen.sh (hunk 1 above)
Comment (MalteHerrmann) [reply scripts/protocgen.sh]: No purpose, thank you for pointing this out

**Comment by MalteHerrmann** [comment]
In discussion with @fedekunze we have decided to ditch the cosmos queries from the swagger-ui docs for now, and to re-add them later on. 
They had to be temporarily removed, because we don't want to store copies of third party proto files anymore, but the `proto-swagger-gen.sh` script depends on these.

**Review by Vvaradinov — APPROVED** [review]
Looks good just wondering if we need to remove `buf.gen.swagger.yaml` from this PR if we won't address the docs generation?

**Comment by MalteHerrmann** [comment]
> Looks good just wondering if we need to remove `buf.gen.swagger.yaml` from this PR if we won't address the docs generation?

We are still adjusting/updating the `proto-swagger-gen.sh` file and this uses the `buf.gen.swagger.yaml`

**Comment by fedekunze** [comment]
@facs95 can you test too?

**Comment by MalteHerrmann** [comment]
> Getting this error when running the command
> 
> ```shell
> Failed to "generate": proto/buf.gen.gogo.yaml has no version set. Please add "version: v1beta1".
> make: *** [proto-gen] Error 1
> ```

That really is weird, because there is a version set: https://github.com/evmos/ethermint/blob/fix-protogen-scripts/proto/buf.gen.gogo.yaml#L1

I think `v1beta1` is something that pops up on the cosmos side (e.g. here: https://github.com/evmos/ethermint/tree/main/third_party/proto/cosmos)? There seems to be a mixup happening somewhere, which I cannot reproduce..

**Comment by itsdevbear** [comment]
works fine 5 me: https://github.com/evmos/ethermint/pull/1465/files#diff-3ee522e37b200be57ea7841f2b3e130313eb7ef78d2c9f7c782178717ac1619f

**Comment by facs95** [comment]
> @facs95 can you test too?

It works fine for me!

**Review by facs95 — APPROVED** [review]

**Comment by MalteHerrmann** [comment]
What do you think @fedekunze, can we merge this? I'm not too sure on how to approach the problems you are having locally, as we have compared the buf and docker versions, which were the same, and there were no other reported failures for other testers.

**Review by fedekunze — APPROVED** [review]
ACK. I had to remove the old images. For some reason was running with the deprecated ones. It would be helpful to add that as a comment on the Makefile

**Pull request merged**